lib/Runtime/Library/JavascriptRegularExpression.h

```diff
@@ -6,6 +6,11 @@
 
 namespace Js
 {
+    struct RegExpTestCache
+    {
+        bool result;
+        Field(JavascriptString*) input;
```

Comment:
What do you think about making this a weak reference to the string?

Reply:
I originally tried this and had problems with this due to ArenaLiteralStrings. These are only used in a couple of places and really don't seem that useful, so I decided to remove them altogether and replace them with normal recycler managed strings.